indra/llcommon/llsdserialize.cpp

```diff
@@ -2177,7 +2180,9 @@ std::string zip_llsd(LLSD& data)
 		{ //copy result into output
 			if (strm.avail_out >= CHUNK)
 			{
-				free(output);
+				deflateEnd(&strm);
+				if(output)
+					free(output);
```

Comment:
`free(nullptr)` is [documented as a no-op](https://en.cppreference.com/w/cpp/memory/c/free).

That said, with various early exits having to remember to `free(output)` and especially with the `realloc()` and copy logic, this looks like a job for `std::vector<U8>` and its handy `resize()` method.